catalog/dags/providers/provider_dag_factory.py

```diff
@@ -537,18 +563,49 @@ def _build_partitioned_ingest_workflows(
     return partitioned_workflows, total_ingestion_metrics
 
 
+@task
+def report_aggregate_reingestion_errors(
+    provider_conf: ProviderReingestionWorkflow, dag_run=None
+):
+    """
+    Report ingestion errors that occurred during a reingestion workflow in
+    a single, aggregate slack message. If no errors were encountered, skip.
+    """
+    # Get the list of failed `pull_data` tasks
+    media_type_name = _get_media_type_descriptor(provider_conf.media_types)
+    failed_pull_data_tasks = [
+        task
+        for task in dag_run.get_task_instances(state=State.FAILED)
+        if f"pull_{media_type_name}_data" in task.task_id
+    ]
+
+    if not failed_pull_data_tasks:
+        raise AirflowSkipException
+
+    message = (
+        f"Ingestion errors were encountered in {len(failed_pull_data_tasks)}"
+        f" ingestion days while running the `{provider_conf.dag_id}` DAG. See the following"
+        " logs for details:\n"
+    ) + "\n".join(
+        f"  - <{task.log_url}|{task.task_id}>" for task in failed_pull_data_tasks
```

Comment:
It would be really cool if we could see the actual exceptions themselves here as well...but I think that would require a lot more code to pull the context or something similar. Ah well, way better than 100s of messages with the same content!

Reply:
Agreed, but yeah I think that would require more work -- and decisions about how to present that in the message without visually cluttering it up again, particularly if not all of the tasks have the same error or whatever.

That would be a really nice addition though for the future... maybe we could have a longer report that groups failed tasks by their error, that's only logged in the aggregate reporting step (rather than being part of the slack message)... 🤔

Reply:
Oooo, that would be such a cool idea! But yea, not necessary here, just pondering!